### PR TITLE
Time Cluster Addition and Colour Coding of Straw Hits

### PR DESCRIPTION
**Pull request by NamithaChitrazee**
The following additions and changes have been made :

- The option to include the "Time Cluster" in the display. It can be set in prolog.fcl like the other options.
- The straw hits are colour coded depending on whether they are part of the reconstructed helix. The option to view the colour coded hits is present in prolog.fcl as well. 
- Some modifications have been made with regard to the MC truth display but it is still not completely ready for viewing.

### CODE REVIEW

File: inc/CollectionFiller.hh
Comment (sophiemiddleton) [inline inc/CollectionFiller.hh:33]: Is there a reason you switched?

File: src/REveMu2eDataInterface.cc
Comment (sophiemiddleton) [inline src/REveMu2eDataInterface.cc:140]: we should avoid hardcoded numbers - perhaps create a "markerstyle" and "markersize" parameter in the header and then use that here

**Review by sophiemiddleton — CHANGES_REQUESTED** [review]
Namitha, this is mostly very good, just a few comments on the hard coded numbers

File: inc/CollectionFiller.hh
Comment (NamithaChitrazee) [reply inc/CollectionFiller.hh:33]: Yes, I was comparing it with the TEve code and its given as fhicl::Atom and not Sequence for the MCTrajectoryCollection. Since, there was an error with the MC truth helix display I was editing the code to be similar to the one in TEve.  

I have also made some other additions/changes like :

Screenshot 2022-01-05 at 15 44 33

Screenshot 2022-01-05 at 15 51 57

So, now there is no Segmentation Fault ( which used to be present earlier) when I was trying to run the display, even when "AddMCTraj" is set to false. But, we need to look into this more to get the MC truth display up and running.

File: src/REveMu2eDataInterface.cc
Comment (NamithaChitrazee) [reply src/REveMu2eDataInterface.cc:140]: Thank you. I will change it.

Comment (sophiemiddleton) [reply src/REveMu2eDataInterface.cc:140]: Great

File: inc/CollectionFiller.hh
Comment (sophiemiddleton) [reply inc/CollectionFiller.hh:33]: Yes, I am going to work on this next. I am trying to change everything into Sequences eventually so we can have multiple collections plotted. It's fine for now but once this is merged I suggest you pull the latest code and work from that. I have opened another PR which I will merge after yours you can start from comparing your work to what's in there and then add you corrections on top. I've not tested it yet at all so it will probably still seg fault, I will look into that in the coming days once both our PR's are merged in. You are welcome to change anything I've done.

File: src/REveMu2eDataInterface.cc
Comment (NamithaChitrazee) [reply src/REveMu2eDataInterface.cc]: I have made the changes.

Comment (sophiemiddleton) [reply src/REveMu2eDataInterface.cc]: Excellent thanks

File: src/REveMu2eDataInterface.cc
Comment (sophiemiddleton) [inline src/REveMu2eDataInterface.cc:141]: This looks like a typo

**Review by sophiemiddleton — COMMENTED** [review]
Almost there, I see a typo - can you correct and just check it builds and works OK on a simple example?

File: src/REveMu2eDataInterface.cc
Comment (NamithaChitrazee) [reply src/REveMu2eDataInterface.cc:141]: Oh yes, I am sorry. Will fix it right away.

Comment (NamithaChitrazee) [reply src/REveMu2eDataInterface.cc]: Hi, it is fixed and I ran it on an art file. It is working. Sorry I did not notice it earlier.

Comment (sophiemiddleton) [reply src/REveMu2eDataInterface.cc]: Great, tjanks for chacking

**Pull request merged**
